### PR TITLE
WebHost: allow newlines in data-tooltip

### PR DESCRIPTION
**Pull request by Berserker66**
Turns 
image
into 
image

### CODE REVIEW

**Comment by black-sliver** [comment]
This makes short tooltips look unexpected:
width

Using max-wdith instead of width should work:
max-width

File: WebHostLib/static/styles/tooltip.css [hunk 1]
```diff
@@ -57,12 +57,13 @@ give it one of the following classes: tooltip-left, tooltip-right, tooltip-top,
 .tooltip:after, [data-tooltip]:after {
 	z-index: 10000;
 	padding: 8px;
-	width: 160px;
+	width: 260px;
```
Comment (black-sliver) [inline WebHostLib/static/styles/tooltip.css:60]:
```suggestion
	max-width: 260px;
```

**Comment by Jarno458** [comment]
POGGERS

File: WebHostLib/static/styles/tooltip.css (hunk 1 above)
Comment (Berserker66) [reply WebHostLib/static/styles/tooltip.css:60]: unfortunately removing width makes the ones on /generate really short and messy.

File: WebHostLib/static/assets/player-settings.js [hunk 2]
```diff
@@ -101,11 +101,16 @@ const buildOptionsTable = (settings, romOpts = false) => {
...
-    label.setAttribute('for', setting);
-    label.setAttribute('data-tooltip', settings[setting].description);
     label.innerText = `${settings[setting].displayName}:`;
+    label.setAttribute('for', setting);
+    const question_span = document.createElement('span');
+    question_span.setAttribute('class', 'interactive')
+    question_span.innerText = ` (?)`;
+
     tdl.appendChild(label);
+    label.appendChild(question_span);
```
Comment (black-sliver) [inline WebHostLib/static/assets/player-settings.js:113]: i think you want regular single quotes there.

Also here you have
`<label>...<span> (?)</span></label>` while in generate you have
`<label>...</label> <span>(?)</span>`

**Comment by black-sliver** [comment]
I noticed two things in the latest iteration:
There is no space between `:` and `(?)` on player-settings.
The mouse cursor changes to "pointer" for the table cell, but back to normal for the `<label>` and `(?)` - until there is something happening on click, i think the pointer is misleading anyway

**Comment by LegendaryLinux** [comment]
I've tried to find a size where the tooltips look nice with a lot of content, but are also small enough to not look weird when only a few words are present. I also fixed the lack of spaces and removed the misleading `cursor:pointer`. Additionally, I changed the target of the tooltip to the span containing the (?) so the tooltip target arrow no longer points at nothing in wide labels with little text.

**Review by LegendaryLinux — APPROVED** [review]
The combination of newlines with the slight adjustment to tooltip width makes them look a lot better.

**Comment by black-sliver** [comment]
> I've tried to find a size where the tooltips look nice with a lot of content, but are also small enough to not look weird when only a few words are present.

In case you missed it, to me
width
(from https://github.com/ArchipelagoMW/Archipelago/pull/921#issuecomment-1216220846 ) looks weird

Not providing the width (or using max-width instead) will result in perfect fit for single line and ~230px for multi-line ones.

**Comment by LegendaryLinux** [comment]
That screenshot does not seem to be from the latest version. Currently, there are yellow question marks on the `/player-settings` pages. Not setting a `min-width` or a `width` results in the tooltip inheriting the width of the parent span, which is very small.

**Comment by black-sliver** [comment]
Current main, bubble width = 176px




Berserker's commit d3d0a77, bubble width = dynamic, max 204px on this specific page




Latest commit 87216b7, bubble width = 276px

**Comment by LegendaryLinux** [comment]
The bubbles were able to be dynamic before because they were targeted on the `<td>` instead of the `<span>`. Their parent element was larger then, but because they are now pointed at the span, their parent element is much smaller and they therefore are very thin due to width inheritance. `min-width` effectively just sets a width now, and setting a `max-width` never allows them to grow. A static width seemed the appropriate solution to this. I don't think there is one set width that is perfect for every option, but we can certainly keep tweaking things.

The reason the new target is the span is because the tooltip arrow would often point at nothing when assigned to the td.

I'm trying to figure out if there is a way to get dynamic width working again. If you have any suggestions, I'm open to hear them.

**Comment by LegendaryLinux** [comment]
Thus far I have not had success re-implementing dynamic width due to the new inherited width. I'm open to suggestions, but would also like to get this merged before the next release.

**Comment by black-sliver** [comment]
I'm not experienced enough with css to make a proper assessment, but i said earlier that i think it is impossible to freely place the tooltip with the current dom structure. The tooltip would have to live outside of the `<td>` for that. You *could* pop it with js, add it to body and position it using js, but having a fallback is still good.

I *personally* would prefer to have all tooltips/bubbles at the same X position for readability (as it happens when aligning it to the `<td>`) rather than being centered on the `(?)`, but that would also mean the triangle/arrow/pointer is off-center, which ... possibly could be done by having the arrow as a different element than the bubble itself (i.e. `(?)` produces the arrow, and `<td>` produces the bubble).

It's not the end of the world and I'm no web dev, so this is just my personal feeling about this.

**Comment by Ijwu** [comment]
Ok, then in the case of @black-sliver not feeling particularly strongly... Merging.

**Pull request merged**